lib/web3-contracts.js

```diff
@@ -40,83 +45,53 @@ export function useKnownContract(name, signer = true) {
 }
 
 export function useTokenDecimals(symbol) {
-  const [decimals, setDecimals] = useState(-1)
-  const tokenContract = useKnownContract(`TOKEN_${symbol}`)
-
-  useEffect(() => {
-    let cancelled = false
-    setDecimals(-1)
-
-    if (symbol === 'ETH') {
-      setDecimals(18)
-    } else if (tokenContract) {
-      tokenContract.decimals().then(decimals => {
-        if (!cancelled) {
-          setDecimals(decimals)
-        }
-      })
-    }
-
-    return () => (cancelled = true)
-  }, [symbol, tokenContract])
-
-  return decimals
+  return tokenDecimals.get(symbol)
 }
 
-export function useEthBalance() {
-  const { account, web3ReactContext } = useWeb3Connect()
-  const [balance, setBalance] = useState(bigNum(-1))
+export function useAnjTotalSupply() {
+  const [loading, setLoading] = useState(false)
+  const [totalSupply, setTotalSupply] = useState(bigNum(-1))
+  const anjContract = useKnownContract('TOKEN_ANJ')
 
-  const cancelBalanceUpdate = useRef(null)
-
-  const updateBalance = useCallback(() => {
+  useEffect(() => {
     let cancelled = false
+    let retryTimer
 
-    if (cancelBalanceUpdate.current) {
-      cancelBalanceUpdate.current()
-      cancelBalanceUpdate.current = null
-    }
-
-    if (!account || !web3ReactContext) {
-      setBalance(bigNum(-1))
+    if (!anjContract) {
+      setTotalSupply(bigNum(-1))
+      setLoading(false)
       return
     }
 
-    cancelBalanceUpdate.current = () => {
-      cancelled = true
-    }
+    const updateTotalSupply = async () => {
+      try {
+        setLoading(true)
 
-    web3ReactContext.library.getBalance(account).then(balance => {
-      if (!cancelled) {
-        setBalance(balance)
-      }
-    })
-  }, [account, web3ReactContext])
+        const totalSupply = await anjContract.totalSupply()
 
-  useEffect(() => {
-    // Always update the balance if updateBalance() has changed
-    updateBalance()
-
-    if (!web3ReactContext || !account) {
-      return
-    }
-
-    const onTransfer = (from, to, value) => {
-      if (from === account || to === account) {
-        updateBalance()
+        if (!cancelled) {
+          setLoading(false)
+          setTotalSupply(totalSupply)
+        }
+      } catch (err) {
+        if (!cancelled) {
+          retryTimer = setTimeout(updateTotalSupply, RETRY_EVERY)
+        }
       }
     }
-    web3ReactContext.library.on('Transfer', onTransfer)
+
+    updateTotalSupply()
 
     return () => {
-      web3ReactContext.library.removeListener('Transfer', onTransfer)
+      cancelled = true
+      clearTimeout(retryTimer)
     }
-  }, [account, web3ReactContext, updateBalance])
+  }, [anjContract])
 
-  return balance
+  return useMemo(() => ({ loading, totalSupply }), [loading, totalSupply])
 }
 
-export function useTokenBalance(symbol) {
+export function useTokenBalance(symbol, address = '') {
```

Comment:
Similar to above, let's just inline any web3 calls we need to get the token balance for the ANJ or ANT balance in `useBondingCurvePrice()`.